OpenRA.Mods.Common/Traits/Cargo.cs

```diff
@@ -203,6 +203,9 @@ public IEnumerable<IOrderTargeter> Orders
 				if (IsTraitDisabled)
 					yield break;
 
+				if (IsEmpty())
```

Comment:
You now duplicate a lot of the code below.

I would just assign a var priority and use that below.